picard/acoustid/__init__.py

```diff
@@ -203,7 +211,12 @@ def _on_fpcalc_finished(self, task, exit_code, exit_status):
         try:
             self._running -= 1
             self._run_next_task()
-            if exit_code == 0 and exit_status == QtCore.QProcess.ExitStatus.NormalExit:
+            # fpcalc returns the exit code 3 in case of decoding errors that
+            # still allowed it to calculate a result.
+            if exit_code in (FpcalcExit.NOERROR, FpcalcExit.DECODING_ERROR) and exit_status == QtCore.QProcess.ExitStatus.NormalExit:
```

Comment:
```suggestion
            if exit_code in {FpcalcExit.NOERROR, FpcalcExit.DECODING_ERROR} and exit_status == QtCore.QProcess.ExitStatus.NormalExit:
```

Set vs tuple (in practice it shouldn't make much difference though).

Reply:
@zas fixed